lib/models/instrumentation.js

```diff
@@ -0,0 +1,277 @@
+      'invoke ember via the local executable within node_modules.  Init ' +
+      'instrumentation will still be recorded, but some bootstraping will be ' +
+      'omitted.'
+    ));
+  }
+}
+
+module.exports = Instrumentation;
+
+Instrumentation.prototype._buildSummary = function (tree, result, resultAnnotation) {
+  var buildSteps = 0;
+  var totalTime = 0;
+
+  var node;
+  var statName;
+  var statValue;
+  var nodeItr;
+  var statsItr;
+  var nextNode;
+  var nextStat;
+
+  for (nodeItr = tree.preOrderIterator();;) {
+    nextNode = nodeItr.next();
+    if (nextNode.done) { break; }
+
+    node = nextNode.value;
+    if (node.label.broccoliNode && !node.label.broccoliCachedNode) {
+      ++buildSteps;
+    }
+
+    for (statsItr = node.statsIterator();;) {
+      nextStat = statsItr.next();
+      if (nextStat.done) { break; }
+
+      statName = nextStat.value[0];
+      statValue = nextStat.value[1];
+
+      if (statName === 'time.self') {
+        totalTime += statValue;
+      }
+    }
+  }
+
+  var summary = {
+    build: {
+      type: resultAnnotation.type,
+      count: this.instrumentations.build.count,
+      outputChangedFiles: result.outputChanges,
+    },
+    output: result.directory,
+    totalTime: totalTime,
+    buildSteps: buildSteps,
+  };
+
+  if (resultAnnotation.type === 'rebuild') {
+    summary.build.primaryFile = resultAnnotation.primaryFile;
+    summary.build.changedFileCount = resultAnnotation.changedFiles.length;
+    summary.build.changedFiles = resultAnnotation.changedFiles.slice(0, 10);
+  }
+
+  return summary;
+};
+
+function totalTime(tree) {
+  var totalTime = 0;
+  var nodeItr;
+  var node;
+  var statName;
+  var statValue;
+  var statsItr;
+  var nextNode;
+  var nextStat;
+
+  for (nodeItr = tree.preOrderIterator();;) {
+    nextNode = nodeItr.next();
+    if (nextNode.done) { break; }
+
+    node = nextNode.value;
+
+    for (statsItr = node.statsIterator();;) {
+      nextStat = statsItr.next();
+      if (nextStat.done) { break; }
+
+      statName = nextStat.value[0];
+      statValue = nextStat.value[1];
+
+      if (statName === 'time.self') {
+        totalTime += statValue;
+      }
+    }
+  }
+
+  return totalTime;
+}
+
+Instrumentation.prototype._initSummary = function (tree) {
+  return {
+    totalTime: totalTime(tree)
+  };
+};
+
+Instrumentation.prototype._commandSummary = function (tree, commandName, commandArgs) {
+  return {
+    name: commandName,
+    args: commandArgs,
+    totalTime: totalTime(tree)
+  };
+};
+
+Instrumentation.prototype._shutdownSummary = function (tree) {
+  return {
+    totalTime: totalTime(tree)
+  };
+};
+
+Instrumentation.prototype._instrumentationFor = function (name) {
+  var instr = this.instrumentations[name];
+  if (!instr) {
+    throw new Error('No such instrumentation "' + name + '"');
+  }
+  return instr;
+};
+
+Instrumentation.prototype._instrumentationTreeFor = function(name) {
+  return tree.loadFromNode(this.instrumentations[name].node);
```

Comment:
` this.project.instrumentation.start('myaddonThingy').` already throws via `instrumentationFor`, which is tested: https://github.com/ember-cli/ember-cli/blob/04cbe974a19beb61af25415901cd9ab78d08dbb4/tests/unit/models/instrumentation-test.js#L325-L330